File: TransCoder_tokenized_test_set_functions/python/PRINT_ALL_DISTINCT_CHARACTERS_OF_A_STRING_IN_ORDER_3_METHODS_1.py
def printDistinct ( Str ) :
    n = len ( Str )
    count = [ 0 for i in range ( MAX_CHAR ) ]
    index = [ n for i in range ( MAX_CHAR ) ]
    for i in range ( n ) :
        x = ord ( Str [ i ] )
        count [ x ] += 1
        if ( count [ x ] == 1 and x != ' ' ) :
            index [ x ] = i
        if ( count [ x ] == 2 ) :
            index [ x ] = n
    index = sorted ( index )
    for i in range ( MAX_CHAR ) :
        if index [ i ] == n :
            break
        print ( Str [ index [ i ] ], end = "" )
    

File: TransCoder_tokenized_test_set_functions/java/PRINT_ALL_DISTINCT_CHARACTERS_OF_A_STRING_IN_ORDER_3_METHODS_1.java
static void printDistinct ( String str ) {
  int n = str . length ( );
  int [ ] count = new int [ MAX_CHAR ];
  int [ ] index = new int [ MAX_CHAR ];
  for ( int i = 0;
  i < MAX_CHAR;
  i ++ ) {
    count [ i ] = 0;
    index [ i ] = n;
  }
  for ( int i = 0;
  i < n;
  i ++ ) {
    char x = str . charAt ( i );
    ++ count [ x ];
    if ( count [ x ] == 1 && x != ' ' ) index [ x ] = i;
    if ( count [ x ] == 2 ) index [ x ] = n;
  }
  Arrays . sort ( index );
  for ( int i = 0;
  i < MAX_CHAR && index [ i ] != n;
  i ++ ) System . out . print ( str . charAt ( index [ i ] ) );
}